MapCanvas.py

```diff
@@ -125,6 +125,12 @@ def __init__(self, master, stage_data=None):
         self.canvas.bind('<B1-Motion>', self.drag_block)
         self.canvas.bind('<ButtonPress-3>', self._show_popup)
 
+        self.canvas.focus_set() # need to set focus for it to detect keyboard inputs
```

Comment:
This will only set the focus when the window is intialised. So essentially this command will do nothing I think.
When you click on something else to edit it the focus will be set there instead.
It should be possible to have it so that if the mouse if over the canvas then the `self.scroll_<dir>` functions will scroll the canvas.